src/murmur/Messages.cpp

```diff
@@ -462,8 +464,27 @@ void Server::msgBanList(ServerUser *uSource, MumbleProto::BanList &msg) {
 				b.qdtStart = QDateTime::currentDateTime().toUTC();
 			}
 			b.iDuration = be.duration();
-			if (b.isValid())
+			if (b.isValid()) {
 				qlBans << b;
+				if (! tmpBans.contains(b)) {
+					log(uSource, QString("New or changed ban: %1, Host: %2, Username: %3, Reason: %4, Length: %5").arg(
+						b.qsHash,
+						b.haAddress.toString(),
+						b.qsUsername,
+						b.qsReason,
+						b.iDuration == 0 ? "Permanent" : "Temporary"));
+				}
+			}
+		}
+		foreach(const Ban &tmpBan, tmpBans) {
+			if (! qlBans.contains(tmpBan)) {
+				log(uSource, QString("Removed ban: %1, Host: %2, Username: %3, Reason: %4, Length: %5").arg(
```

Comment:
Might be worth extracting the "turn a ban into a string" functionality into the Ban class (maybe as ```toString```). Not sure if we need it in any other locations but with just the two locations here it would already simplify things.